File: mayday_control/scripts/structure/structure.py
from geometry.pose import Pose
from geometry.vec3 import Vec3
from side import Side
from structure.components import Joint, Link
from structure.leg_position import LegPosition


class Structure():
    def __init__(self, links: [Link], joints: [Joint]):
        pass

    def get_pose_of(self, ):
        pass


class MaydayStructure(Structure):
    def get_thorax_pose(self) -> Pose:
        return Pose.zeros()
        # TODO fix cheat value

    def get_leg_root_pose(self, leg_position: LegPosition, side: Side) -> Pose:
        return self._get_leg(leg_position, side).origin

    # def _get_leg(self, leg_position: LegPosition, side: Side) -> Leg:
    #     return self._legs[(leg_position, side)]

File: mayday/Structures/Structure.cs
using mayday.Geometry;

namespace mayday.Structures;

public class Structure(IEnumerable<Joint> joints, IEnumerable<Attachment> attachments, IEnumerable<Link> links)
{
    public IEnumerable<JointState> JointStates => joints.Select(j => j.State);

    public Pose GetPoseFor(Component component)
    {
        // TODO fix cheat and actually do forward kinematics
        return component.Origin;
    }
}
